### DISCUSSION TITLE
$refresh() not rollbacking change

### DISCUSSION THREAD

**Discussion by xanadu72** [CodeDredd/pinia-orm#973] [Q&A] [answered]
I try to use use refresh without success : 
```javascript
	it('can refresh to its original state', () => {
		const user = new User({ id: 1, lastName: 'John Doe' })
		user.lastName = 'Johnny Doe'
		user.$refresh()
		expect(user.$isDirty()).toBeFalsy()
		expect(user.lastName,"Refresh not restoring original").toBe('John Doe');
	})
```

Did I misunderstand the role of the refresh function ?

**Comment by CodeDredd** [comment, marked as answer]
I need more details. Can please provide a reproduction please ? 🙏 Thank you!

**Reply by xanadu72** [reply]
I can't provide you more details except the User class but it seems that the lastName is not restored to the original value. You can probably update your own test (Model.spec.ts) and check if the name is restored.

**Reply by CodeDredd** [reply]
fixed with https://github.com/CodeDredd/pinia-orm/releases/tag/pinia-orm%401.6.5
